proptest/src/test_runner/runner.rs

```diff
@@ -611,6 +632,33 @@ impl TestRunner {
         while self.successes < self.config.cases {
             // Generate a new seed and make an RNG from that so that we know
             // what seed to persist if this case fails.
+
+            if self.config.edge_bias <= 0f32 {
+                self.current_edge_bias = 0f32;
+            } else if self.config.edge_bias >= 1f32 {
+                self.current_edge_bias = 1f32;
+            } else {
+                // Value from 0.0 to 1.0 representing how far we are through the
+                // tests. Infinity is OK here in case of cases == 1.
+                let x = self.successes as f32 / (self.config.cases - 1) as f32;
+                if x.is_infinite() {
+                    self.current_edge_bias = self.config.edge_bias;
+                } else {
+                    // Separate the curve into two linear parts whose total area
+                    // is p. Cases of p <= 0 and p >= 1 are handled above.
+                    let p = self.config.edge_bias;
+                    if x < p {
+                        self.current_edge_bias = 1f32 + x * (p - 1f32) / p;
+                    } else {
+                        self.current_edge_bias = (x - 1f32) * p / (p - 1f32);
+                    }
```

Comment:
I think we should add a bit more documentation. maybe something like:
"""
Separate the curve into two linear parts whose total area
is p. Cases of p <= 0 and p >= 1 are handled above.
1. Before x reaches p: Bias decreases from 1.0 toward p.
2. After x reaches p: Bias decreases from p toward 0.0.
The first region ensures that edge cases are mixed into tests. The second region ensures we are testing all cases uniformly.
"""

Reply:
Sounds like a pretty good description.